eventsourcing/src/main/java/org/axonframework/eventsourcing/IEventSourcingHandler.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.eventhandling.EventHandler;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.MessageStream;
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+
+// todo: rename to EventSourcingHandler and move the `@EventSourcingHandler` annotation to annotations package
+@FunctionalInterface
```

Comment:
Don't forget the JavaDoc here when rounding up this PR! ;-)

Reply:
Done, please check the docs.